packages/kit/src/utils/streaming.js

```diff
@@ -23,22 +23,29 @@ function defer() {
  * }}
  */
 export function create_async_iterator() {
-	let deferred = defer();
+	let deferred = [defer()];
 
 	return {
 		iterator: {
 			[Symbol.asyncIterator]() {
 				return {
-					next: () => deferred.promise
+					next: async () => {
+						const next = await deferred[0].promise;
+						if (!next.done) deferred.shift();
+						return next;
+					}
 				};
 			}
 		},
 		push: (value) => {
-			deferred.fulfil({ value, done: false });
-			deferred = defer();
+			deferred[deferred.length - 1].fulfil({
+				value,
+				done: false
+			});
+			deferred.push(defer());
 		},
 		done: () => {
-			deferred.fulfil({ done: true });
+			deferred[deferred.length - 1].fulfil({ done: true });
```

Comment:
```suggestion
			deferred.at(-1).fulfil({ done: true });
```

Reply:
I had that previously but that requires you to do a stupid type cast because TS says "this might be undefined" and that code was harder to read/longer.

Reply:
wait TS thinks `array.at(x)` might be undefined but `array[x]` is not? the fuck?

Reply:
I think this is mostly due to backwards compatibility to not break existing code bases. They have another flag outside the `strict` family that turns on the same behavior for `array[x]`